libs/agno/agno/models/base.py

```diff
@@ -1080,6 +1121,20 @@ def run_function_call(
 
         function_call_success = function_execution_result.status == "success"
 
+        # Handle media artifacts from function results
+        if function_call_success and hasattr(function_call.function, '_agent'):
```

Comment:
No shouldn't do this. Should only add it to function_execution_result and let it bubble up in the same way (via ModelResponse events) as `session_state`